pkg/components/exporter/exporter.go

```diff
@@ -74,16 +76,33 @@ func NewDaemonSet(detail components.Detail, k *v1alpha1.KeplerInternal) *appsv1.
 	}
 
 	deployment := k.Spec.Exporter.Deployment.ExporterDeploymentSpec
-	image := k.Spec.Exporter.Deployment.Image
 	nodeSelector := deployment.NodeSelector
 	tolerations := deployment.Tolerations
-	port := deployment.Port
 	// NOTE: since 2 or more KeplerInternals can be deployed to the same namespace,
 	// we need to make sure that the pod selector of each of the DaemonSet
 	// create of each kepler is unique. Thus the daemonset name is added as
 	// label to the pod
 
-	bindAddress := "0.0.0.0:" + strconv.Itoa(int(port))
+	exporterContainer := newExporterContainer(k.Name, k.DaemonsetName(), k.Spec.Exporter.Deployment)
```

Comment:
How about `newExporterContainer` return an object instead of a pointer since `nil` is never returned?